### PR TITLE
Initial policy for haproxy

### PR DESCRIPTION
**Pull request by 0xC0ncord**
Add an initial policy for haproxy, a TCP/HTTP reverse proxy for high availability environments.

### CODE REVIEW

File: policy/modules/services/haproxy.te [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+type haproxy_tmpfs_t;
+files_tmpfs_file(haproxy_tmpfs_t)
+
+########################################
+#
+# haproxy local policy
+#
+
+allow haproxy_t self:process { getsched setrlimit signal };
+allow haproxy_t self:capability { kill net_admin net_bind_service setuid setgid };
```
Comment (pebenito) [inline policy/modules/services/haproxy.te:65]: The corenet bind interfaces should be providing `net_bind_service` as needed, so I'd prefer not to have it as an explicit allow. If it's not happening, then there is a bug in corenet generation scripts.

File: policy/modules/services/haproxy.te [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+dev_dontaudit_read_sysfs(haproxy_t)
+
+kernel_read_kernel_sysctls(haproxy_t)
+kernel_read_state(haproxy_t)
+kernel_read_system_state(haproxy_t)
+
+auth_use_nsswitch(haproxy_t)
+
+miscfiles_read_generic_certs(haproxy_t)
+
```
Comment (pebenito) [inline policy/modules/services/haproxy.te:86]: Extra newline

File: policy/modules/services/haproxy.te [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+manage_sock_files_pattern(haproxy_t, haproxy_runtime_t, haproxy_runtime_t)
+files_runtime_filetrans(haproxy_t, haproxy_runtime_t, { dir file sock_file })
+
+mmap_manage_files_pattern(haproxy_t, haproxy_tmpfs_t, haproxy_tmpfs_t)
+fs_tmpfs_filetrans(haproxy_t, haproxy_tmpfs_t, file)
+
+create_dirs_pattern(haproxy_t, haproxy_log_t, haproxy_log_t)
+create_files_pattern(haproxy_t, haproxy_log_t, haproxy_log_t)
+append_files_pattern(haproxy_t, haproxy_log_t, haproxy_log_t)
+logging_log_filetrans(haproxy_t, haproxy_log_t, { dir file })
```
Comment (pebenito) [inline policy/modules/services/haproxy.te:105]: Needs to move to line 70 and put log access block after conf access.

File: testing/sechecker.ini [hunk 4]
```diff
@@ -254,6 +254,7 @@ exempt_source = arpwatch_t
                 drbd_t
                 fcoemon_t
                 firewalld_t
+                haproxy_t
```
Comment (pebenito) [inline testing/sechecker.ini:257]: Please put a comment for why/how haproxy needs net_admin.

**Review by pebenito — CHANGES_REQUESTED** [review]

File: testing/sechecker.ini (hunk 4 above)
Comment (0xC0ncord) [reply testing/sechecker.ini:257]: I found that haproxy is still functioning in my setup if I `dontaudit` the access, so this won't be needed anymore. :)

File: policy/modules/services/haproxy.te (hunk 1 above)
Comment (0xC0ncord) [reply policy/modules/services/haproxy.te:65]: Neat. Frankly I never knew these interfaces implied that access. I tested without `net_bind_service` explicitly given and it still has that access, so the interfaces seem to be working as intended.

**Review by pebenito — APPROVED** [review]

**Pull request merged**